activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -501,6 +510,21 @@ def combine_bind_parameters(
 
       private
 
+        def with_connection_verify(exception_class)
+          yield
+        rescue exception_class
+          raise_unless_verifying_connection
+          reconnect!
+          retry
```

Comment:
Will this retry `INSERT` or `UPDATE` queries?  If so, the connection could get lost between when the query is committed and the response is received.  That means we could end up double inserting.